File: investpulse/frontend/src/types/market.ts
export interface Candle {
  timestamp: string
  open: number
  high: number
  low: number
  close: number
  volume: number
}

export interface Ticker {
  market: string
  trade_price: number
  signed_change_rate: number
  acc_trade_volume_24h: number
  high_price: number
  low_price: number
  timestamp: number
}

export interface MarketInfo {
  market: string
  korean_name: string
  english_name: string
}

export interface RealtimePrice {
  market: string
  trade_price: number
  signed_change_rate: number
  acc_trade_volume_24h: number
  trade_timestamp: number
}

File: investpulse/backend/app/schemas/market.py
from datetime import datetime

from pydantic import BaseModel


class CandleResponse(BaseModel):
    timestamp: str
    open: float
    high: float
    low: float
    close: float
    volume: float


class TickerResponse(BaseModel):
    market: str
    trade_price: float
    signed_change_rate: float
    acc_trade_volume_24h: float
    high_price: float
    low_price: float
    timestamp: int


class MarketInfo(BaseModel):
    market: str
    korean_name: str
    english_name: str


class IndicatorResponse(BaseModel):
    rsi: dict
    macd: dict
    bollinger_bands: dict
    timestamps: list[str]
